tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,969 @@
+     'Comma-separated list of upload IDs to delete. All the uploads must be associated to the same archive.'],
+    ['-protocol', 'const', 1, \$deleteProtocols,
+     'Delete the entries in mri_processing_protocols tied to the upload(s) specified via -uploadID.'],
+    ['-form', 'const', 1, \$deleteMriParameterForm,
+     'Delete the entries in mri_parameter_form tied to the upload(s) specified via -uploadID.']
+);
+
+my $Help = <<HELP;
+HELP
+
+my $usage = <<USAGE;
+Usage: $0 [-profile profile] [-ignore] [-nobackup] [-protocol] [-form] [-uploadID uploadIDList]
+USAGE
+
+&Getopt::Tabular::SetHelp($Help, $usage);
+&Getopt::Tabular::GetOptions(\@opt_table, \@ARGV) || exit $NeuroDB::ExitCodes::GETOPT_FAILURE;
+
+if(@ARGV != 0) {
+    print STDERR "$usage\n";
+    exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+if(!defined $uploadIDList) {
+    print STDERR "Missing -uploadID option\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+
+# Split the comma-separated string into a list of numbers
+my @uploadID = split(',', $uploadIDList);
+
+# Eliminate duplicate IDs
+my %uploadID = map { $_ => 1 } @uploadID;
+@uploadID = keys %uploadID;
+
+
+#======================================#
+# Validate all command line arguments  #
+#======================================#
+if(grep(!/\d+/, @uploadID)) {
+    print STDERR "Argument to -uploadID option has to be a comma-separated list of numbers\n";
+    exit $NeuroDB::ExitCodes::MISSING_ARG;
+}
+
+if (!$ENV{LORIS_CONFIG}) {
+    print STDERR "\n\tERROR: Environment variable 'LORIS_CONFIG' not set\n\n";
+    exit $NeuroDB::ExitCodes::INVALID_ENVIRONMENT_VAR; 
+}
+
+if (!-e "$ENV{LORIS_CONFIG}/.loris_mri/$profile") {
+    print $Help; 
+    print STDERR "Cannot read profile file '$ENV{LORIS_CONFIG}/.loris_mri/$profile'\n";  
+    exit $NeuroDB::ExitCodes::PROFILE_FAILURE;
+}
+
+# Incorporate contents of profile file
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+
+if ( !@Settings::db ) {
+    print STDERR "ERROR: You don't have a \@db setting in file "
+                 . "$ENV{LORIS_CONFIG}/.loris_mri/$profile";
+    exit $NeuroDB::ExitCodes::DB_SETTINGS_FAILURE;
+}
+
+#====================================#
+# Establish database connection.     #
+#====================================#
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+# Die as soon as a DB operation fails. As a side-effect, any transaction that has not
+# been commited at that point will automatically be rolled back
+$dbh->{'RaiseError'} = 1;
+
+# Do not print an error message when a DB error occurs since the error message 
+# will be printed when the program dies anyway 
+$dbh->{'PrintError'} = 0;
+
+my $dataDirBasepath    = NeuroDB::DBI::getConfigSetting(\$dbh,'dataDirBasepath');
+$dataDirBasepath =~ s!/$!!;
+my $tarchiveLibraryDir = NeuroDB::DBI::getConfigSetting(\$dbh,'tarchiveLibraryDir');
+$tarchiveLibraryDir =~ s!/$!!;
+
+my $query = "SELECT m.UploadID, m.TarchiveID, t.ArchiveLocation, m.SessionID "
```

Comment:
mmmm. I just realized that if you have an `mri_upload` entry that is not linked to a tarchiveID (MRI pipeline not run on it yet or that failed at the dicomTar level) then you get the following error message:
`No upload found in table mri_upload with upload ID 158` but there is indeed an entry in mri_upload but with TarchiveID=NULL